static/app/actionCreators/tags.tsx

```diff
@@ -192,3 +208,214 @@ export function fetchSpanFieldValues({
     query,
   });
 }
+
+type FetchOrganizationTagsParams = {
+  orgSlug: string;
+  // TODO: Change this to Dataset type once IssuePlatform is added
+  dataset?: string;
+  projectIds?: string[];
+  useCache?: boolean;
+};
+
+export const makeFetchOrganizationTags = ({
+  orgSlug,
+  dataset,
+  projectIds,
+  useCache = true,
+}: FetchOrganizationTagsParams): ApiQueryKey => [
+  `/organizations/${orgSlug}/tags/`,
+  {query: {dataset, useCache, project: projectIds}},
+];
+
+export const useFetchOrganizationTags = (
+  params: FetchOrganizationTagsParams,
+  options: Partial<UseApiQueryOptions<Tag[]>>
+) => {
+  return useApiQuery<Tag[]>(makeFetchOrganizationTags(params), {
+    staleTime: Infinity,
+    keepPreviousData: true,
```

Comment:
For hooks in this file we'd like for them to be widely reusable, so I wouldn't add specific config like `keepPreviousData` here, I'd use that config when this gets consumed if the use case requires it.